android/src/main/java/com/reactnativestripesdk/Mappers.kt

```diff
@@ -419,34 +419,39 @@ internal fun mapToAddress(addressMap: ReadableMap?, cardAddress: Address?): Addr
 }
 
 internal fun mapToBillingDetails(billingDetails: ReadableMap?, cardAddress: Address?): PaymentMethod.BillingDetails? {
-  if (billingDetails == null) {
-    return null
+  if (billingDetails == null && cardAddress == null) {
+    return null;
   }
   val address = Address.Builder()
-    .setPostalCode(getValOr(billingDetails, "addressPostalCode"))
-    .setCity(getValOr(billingDetails, "addressCity"))
-    .setCountry(getValOr(billingDetails, "addressCountry"))
-    .setLine1(getValOr(billingDetails, "addressLine1"))
-    .setLine2(getValOr(billingDetails, "addressLine2"))
-    .setState(getValOr(billingDetails, "addressState"))
-
-    cardAddress?.let { ca ->
-      ca.postalCode?.let {
-        if (it.isNotEmpty()) {
-          address.setPostalCode(it)
-        }
-      }
-      ca.country?.let {
-        address.setCountry(it)
+  if (billingDetails != null) {
+    address
```

Comment:
if we move 
```
  val billingDetailsBuilder =  PaymentMethod.BillingDetails.Builder()
```
up to where we're also declaring the other builder, we can consolidate the two `if (billingDetails != null) {` blocks down to one. 

maybe something like:

```
  val billingDetailsBuilder =  PaymentMethod.BillingDetails.Builder()
  val addressBuilder = Address.Builder()

  if (billingDetails != null) {
    billingDetailsBuilder
      .setName(getValOr(billingDetails, "name"))
      .setPhone(getValOr(billingDetails, "phone"))
      .setEmail(getValOr(billingDetails, "email"))
    addressBuilder
      .setPostalCode(getValOr(billingDetails, "addressPostalCode"))
      .setCity(getValOr(billingDetails, "addressCity"))
      .setCountry(getValOr(billingDetails, "addressCountry"))
      .setLine1(getValOr(billingDetails, "addressLine1"))
      .setLine2(getValOr(billingDetails, "addressLine2"))
      .setState(getValOr(billingDetails, "addressState"))
  }
```
?

Reply:
Thanks for spotting the code smell. I'll fix it!